cmake/public/gridtools_setup_targets.cmake

```diff
@@ -218,28 +218,29 @@ macro(_gt_setup_targets _config_mode clang_cuda_mode)
     _gt_add_library(${_config_mode} stencil_naive)
     target_link_libraries(${_gt_namespace}stencil_naive INTERFACE ${_gt_namespace}gridtools)
 
-    set(_required_nlohmann_json_version "3.7.3")
-    if(NOT _nlohmann_json_already_fetched)
-        find_package(nlohmann_json ${_required_nlohmann_json_version})
-    endif()
-    if(NOT nlohmann_json_FOUND)
-        set(_gridtools_repository "https://github.com/GridTools/gridtools.git")
-        set(_gridtools_tag        "v${_required_gridtools_version}")
-        include(FetchContent)
-        FetchContent_Declare(json
-                GIT_REPOSITORY https://github.com/nlohmann/json.git
-                GIT_TAG "v${_required_nlohmann_json_version}")
-        FetchContent_GetProperties(json)
-        if(NOT json_POPULATED)
-            FetchContent_Populate(json)
-            set(JSON_BuildTests OFF CACHE INTERNAL "")
-            add_subdirectory(${json_SOURCE_DIR} ${json_BINARY_DIR} EXCLUDE_FROM_ALL)
+    option(GT_ENABLE_STENCIL_DUMP "Enables the dump backend (with dependencies)" OFF)
```

Comment:
`stencil_dump` is used in regression cmake unconditionally. Needed to be fixed.

The worrying thing is that we handle our optional dependencies differently in the case of `json` and in the case of `hpx`.  I briefly browse the code and figured out that `hpx` thread pool is only enabled in tests if the target `threadpool_hpx` exists. I failed to find the place where `threadpool_hpx` is generated. Is it expected to be external against `gridtools`?

What if we will do the following:
  - our option controls only the fetching of the `nlohmann_json`
  - stencil_dump target is generated if `the nlohmann_json::nlohmann_json` exists (if nlohmann_json is present before gridtools is populated `stencil_dump` will be still available.  
  - `dump` executable is regression is generated only if `stencil_dump` exists

I think it will be more consistent with the approach we take with the `hpx`.

Reply:
makes sense

Reply:
`threadpool_hpx` got lost some time ago. Time for another `GT_TESTS_REQUIRE`...?